### PR TITLE
fix(add_sub) optimize addsubchip and addoperation

### PR DESCRIPTION
**Pull request by perseus34x**
1. Removed the intermediate merged_events vector allocation. The trace generation now indexes directly into add_events or sub_events based on the row index, reducing memory pressure.
2. Removed println! statements inside the hot trace generation loop.
3. The original code only checked shard.add_events to determine if the chip should be active. We corrected this to check shard.add_events.is_empty() || !shard.sub_events.is_empty().
4. Removed the op_a_not_0 column from AddSubCols. It was allocated and populated but never used in the AIR eval function, wasting proof size.
5. Used complex "detect and verify" logic requiring 9+ constraints to check overflows and carries separately. Implemented a direct arithmetic constraint: $A + B + C_{in} = Result + 256 \cdot C_{out}$. Reduced the algebraic complexity to ~3 linear constraints per row, significantly lowering the degree of the constraints.
6. Added Edge Cases for tests.

## PR Checklist

- [X] Added Tests
- [X] Added Documentation
- [ ] Breaking changes

### CODE REVIEW

File: crates/rwasm/executor/src/events/byte.rs [hunk 1]
```diff
@@ -30,6 +30,9 @@ pub struct ByteLookupEvent {
 
 /// A type that can record byte lookup events.
 pub trait ByteRecord {
+    fn is_enabled(&self) -> bool {
```
Comment (djadjka) [inline crates/rwasm/executor/src/events/byte.rs]: I'm not sure that we need it, because _EmptyByteRecord_ already do nothing

Comment (perseus34x) [reply crates/rwasm/executor/src/events/byte.rs]: This mechanism ensures that byte lookups are never generated during generate_trace. Alternatively, we could implement a runtime type check similar to Java's instanceof using blu.as_any().is::<EmptyByteRecord>(). However, this would require converting ByteRecord to a dyn Trait + Any to allow for downcasting, which introduces a small runtime overhead.

Comment (djadjka) [reply crates/rwasm/executor/src/events/byte.rs]:
```
if record.is_enabled() {
    record.add_u8_range_checks(&a);
    record.add_u8_range_checks(&b);
    record.add_u8_range_checks(&expected.to_le_bytes());
}
```
It's only one place when you use it, and it just calls 3 empty functions.

Even if it isn't optimized by the compiler, it's such a small overhead that it's not worth adding another function that adds more questions about where to use it.

Comment (perseus34x) [reply crates/rwasm/executor/src/events/byte.rs]: This is not only for 3 empty functions. for other chips, there are creation of ByteLookupEvent.  I agree with you so i removed is_enabled that brings questions. but i switch ByteRecord plumbing to dyn + Any for runtime type checks, this is more readable

File: crates/rwasm/machine/src/operations/add.rs [hunk 2]
```diff
@@ -25,27 +25,19 @@ impl<F: Field> AddOperation<F> {
...
+            self.carry[i] = F::from_canonical_u16(current_carry);
         }
 
         let base = 256u32;
         let overflow = a[0].wrapping_add(b[0]).wrapping_sub(expected.to_le_bytes()[0]) as u32;
         debug_assert_eq!(overflow.wrapping_mul(overflow.wrapping_sub(base)), 0);
 
-        // Range check
-        {
+        if !record.as_any().is::<EmptyByteRecord>() {
```
Comment (djadjka) [inline crates/rwasm/machine/src/operations/add.rs]: Please don't complicate the code, it doesn't provide any performance improvement.
